server/src/main/java/org/elasticsearch/persistent/PersistentTasksCustomMetaData.java

```diff
@@ -94,8 +92,8 @@ public PersistentTasksCustomMetaData(long lastAllocationId, Map<String, Persiste
         ObjectParser<TaskDescriptionBuilder<PersistentTaskParams>, String> parser = new ObjectParser<>("named");
         parser.declareObject(TaskDescriptionBuilder::setParams,
                 (p, c) -> p.namedObject(PersistentTaskParams.class, c, null), new ParseField("params"));
-        parser.declareObject(TaskDescriptionBuilder::setStatus,
-                (p, c) -> p.namedObject(Status.class, c, null), new ParseField("status"));
+        parser.declareObject(TaskDescriptionBuilder::setState,
```

Comment:
I think we need the status too for bwc? can you see what it takes to have a BWC test which shuts down a cluster with a persistent task and then does a full restart and check it started again with the right state? This can be a follow up.

Reply:
I think it's already here with `new ParseField("state", "status")`? It produces a deprecation warning but allows to parses `status` as it was `state`. Since this is a named object, `status` will be parsed a as `PersistentTaskState` object and this is not an issue.


> can you see what it takes to have a BWC test which shuts down a cluster with a persistent task and then does a full restart and check it started again with the right state?

Sure. When backporting to 6.x I'd like to add a basic bwc unit test fot this status/state renaming and the associated deprecation warning.

Reply:
> I think it's already here with new ParseField("state", "status")? 

Heh. I don't know how I missed it. Sorry for the noise.

> Sure. In addition to this, I'd like to add a basic bwc unit test for this status/state renaming and the associated deprecation warning in 6.x.

I think we can do this as a follow up?